docs/source/method_cards/decoupled_weight_decay.md

```diff
@@ -4,50 +4,68 @@
+from torch.utils.data import DataLoader
+from tests.common import RandomImageDataset
 
-- `lr` - Learning rate.
-- `betas` - Coefficients used for computing running averages of the gradient and its square.
-- `eps` - Term for numerical stability.
-- `weight_decay` - Weight decay.
-- `amsgrad` - Use AMSGrad variant.
+from composer.models import composer_resnet
 
-## Applicable Settings
+model = composer_resnet('resnet50')
 
-Using decoupled weight decay is considered a best practice in most settings. `DecoupledSGDW` and `DecoupledAdamW` should always be used in place of their vanilla counterparts.
-
-## Implementation Details
+train_dataloader = DataLoader(RandomImageDataset(), batch_size=2)
+eval_dataloader = DataLoader(RandomImageDataset(), batch_size=2)
+```
+-->
+<!--pytest-codeblocks:cont-->
+```python
+# Instantiate the optimizer and pass it into the Trainer
+
+from composer.optim import DecoupledSGDW
+from composer.trainer import Trainer
+
+optimizer = DecoupledSGDW(
+    model.parameters(),
+    lr=0.05,
+    momentum=0.9,
+    weight_decay=2.0e-3
+)
+
+trainer = Trainer(model=model,
+                    train_dataloader=train_dataloader,
+                    eval_dataloader=eval_dataloader,
+                    max_duration='1ep',
+                    optimizers=optimizer)
+
+trainer.fit()
+```
 
+### Implementation Details
 Unlike most of our other methods, we do not implement decoupled weight decay as an algorithm, instead providing optimizers that can be used as drop-in replacements for `torch.optim.SGD` and `torch.optim.Adam`; though note that some hyperparameter tuning may be required to realize full performance improvements.
 
 The informed reader may note that PyTorch already provides a `torch.optim.AdamW` variant that implements Loshchilov et al.'s method. Unfortunately, this implementation has a fundamental bug owing to PyTorch's method of handling learning rate scheduling. In this implementation, learning rate schedulers attempt to schedule the weight decay (as Loschilov et al. suggest) by tying it to the learning rate. However, this means that weight decay is now implicitly tied to the initial learning rate, resulting in unexpected behavior where runs with different learning rates also have different effective weight decays. See [this line](https://github.com/pytorch/pytorch/blob/d921891f5788b37ea92eceddf7417d11e44290e6/torch/optim/_functional.py#L125).
 
-## Considerations
+## Suggested Hyperparameters
 
-There are no known negative side effects to using decoupled weight decay once it is properly tuned, as long as the original base optimizer is either `torch.optim.Adam` or `torch.optim.SGD`.
+Optimizers with decoupled weight decay can be used as drop-in replacements for their
+non-decoupled counterparts, however the optimal `weight_decay` value for decoupled
+optimizers will typically be larger than for their non-decoupled counterparts, because
+decoupled weight decay is not scaled by the learning rate.
+## Technical Details
```

Comment:
```suggestion
Optimizers with decoupled weight decay can be used as drop-in replacements for their
non-decoupled counterparts. However, the optimal `weight_decay` value for decoupled
optimizers will typically be larger than for their non-decoupled counterparts, because
decoupled weight decay is not scaled by the learning rate.
## Technical Details
```
Also, is it larger values and not smaller? With an lr of .01 I would expect the weight decay to be equivalent when 100x smaller, but I also didn't think about this too carefully just now.

Reply:
It's definitely smaller, good catch